### DISCUSSION TITLE
Feature Request: Thino取消对<br>的依赖之后，感觉很不好，建议恢复原来的样子

### DISCUSSION THREAD

**Discussion by whong1618** [Quorafind/Obsidian-Thino#517] [Ideas 想法]
### Feature Requested | 请求的功能

闪念笔记类型的Thino作为Obsidian的插件，其优势就是和Obsidian的兼容性和格式设计上基本的一致性。

在Thino没有取消对\<br>的依赖之前，使用感觉很好：
在Thino编辑器编辑的内容，所见即所得，
编辑器中的格式和最后卡片的格式以及在每日Thino里浏览的格式基本保持一致

而且每日笔记最后生成的笔记md文件里，也基本上和卡片保持一致的格式，
不管是笔记md文件的编辑模式，还是阅读模式，都基本一致，与Obsidian其他md笔记编辑体验具有很好的一致性。

然而Thino新版本，取消了对\<br>的依赖，用什么缩进来自动换行。
在Markdown里格式里，缩进是由特殊用途的。

使用Obsidian的Thino做笔记，对我一个大的吸引力就是可以使用Markdown的格式语法来写笔记，所见即所得。而\<br>作为换行，其实很通用，有时候也很好用。

现在的新版本，取消了对\<br>的依赖，在Thino编辑器里，现在两行之间不换你在编辑器中换多少行，最后呈现在Thino卡片里的，都是不认的，间隔2行、3行、4行的原文本，最后呈现的都是一样的间隔。

灵活性失去了，用户编辑器中的所写和最后的呈现出现了格式上的不一致。
笔记md文档，也因为换行\<br>的失去，而在阅读模式里，更是塌缩到了一块。想要体现多个空行，得手动添加\<br>，其实这并不利于用户使用体验。

感觉没有取消对\<br>的依赖之前的设计，使用体验更好，能更好地 与Obsidian格式兼容，同时更好的体现了编辑窗口中的所见即所得。

希望开发者能好好考虑一下，以真正使用者的实际体验以及Thino赖以与对手区别的优势考虑，最好能恢复之前的设计。

### Checklist | 检查清单

- [X] The feature would be useful to more users than just me. 该功能对于不仅仅是我自己的用户来说是有用的。

[a comment by an automated account is not shown]

**Comment by Quorafind** [comment]
之前的设计会导致所有任务相关内容，其它插件无法识别。所有代码块、引用、callout 等，外部插件都无法识别。

**Reply by whong1618** [reply]
@Quorafind  @juestchaos @windily-cloud 

首先感谢 Thino 开发人员的努力，版本 2.4.13及以上 的改善，已经挺符合期待的样子，肯定还有完善的地方，目前看，这次改善是一大进步。

【1】昨晚更新了 2.4.12，针对更新：Thino 取消了对\<br>的依赖，换行使用缩进自动完成，我做了测试，发现：

除了时间数字和 Thino 原卡片笔记内容之间多了些缩进之外，在对应日记的 md 的文件中，引用代码块、callout 等元素在 Osidian 的编辑视图，依然显示源码，而不是解析后的格式，阅读视图的显示也不尽如人意，也是因为发现了这些不满意的地方，所以没有感受到开发者取消 Thino 对\<br>依赖改进的用意和好处。 

（1）【版本2.4.12】测试了一下，笔记对应的日记md文件的编辑视图中多了很多缩进，卡片内容依然显示源码，不显示解析后的格式。
![Thino测试日记md中编辑视图【版本2 4 12】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/e4a37aed-0cdb-49e5-9c4d-75b50c241df8)

（2）【版本2.4.12】测试了一下，笔记对应的日记md文件的阅读视图，堆积到了一块，而且不能正确解析格式
![Thino测试日记md中阅读视图【版本2 4 12】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/79da5deb-554b-4e6b-abfa-f3b3a0c440f2)

【2】今天更新到 2.4.13及以上版本之后，应该是开发人员做了完善。测试了一下，体验到了新版本的好处！

之前的版本，Thino 笔记的内容，会以时间点生成一个无序列表项，由于 Thino 笔记写入日记对应 md 文件时，日记md 文件里的原 Thino 内容会以源码格式列在时间点的无序列表项里，此时，不管是Obsidian的编辑视图还是阅读视图，都不能正确解析大多数格式，只显示源码。

新版本 2.4.13及以上，对这个问题有了很大改善，虽然Thino笔记对应的日记md文件在编辑视图里个别格式比如callout还是不能完全正确解析，但其他多数格式，已经可以做到所见即所得，多数格式得到正确解析。

而在阅读视图里，显示的效果和 每日Thino里 显示的基本一样，这一点是很好的改善，也是我期待的样子。

【Thino笔记对应的日记md文件，在编辑视图里，所见即所得的显示效果略有不足，如果可行的情况下，希望开发者可以予以改善】：
![Thino测试日记md中编辑视图【版本2 4 16】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/28eb93b6-61f2-4e49-9d9c-cda09acd138c)

（1）标题在编辑视图下，没有得到很好的格式解析。
（2）引用文本和前面的标志竖线离得有点远。
（3）代码块在编辑视图倒是能显示，不过整体代码在编辑视图下，多了一个缩进，阅读视图下正常，这点倒影响不大
（4）callout 在 日记 md 里显示虽然成块显示，但是里面显示的依然是源码，而不是 callout 的格式样式，不过在阅读视图倒是可以正常显示。

【3】2.4.13及以上版本，也让我理解了 Thino，取消对\<br>的依赖，这样和 Obsidian自身的编辑器有了更好的兼容性，因为Obsidian本身换行的时候，是不生成\<br>，Thino这次取消对\<br>的依赖，提升了兼容性，这样也体现在Thino笔记对应日记md文件中，编辑视图和阅读视图显示效果的改善。这种改善会提升Thino与其他插件的兼容性，从而更好的使得其他插件，可以识别笔记中的任务列表、代码块和 callout 等元素。

【对于Thino对应日记md文件在编辑视图下，标题、引用、代码块及callout等元素目前格式显示和解析还存在一些不足，有待进一步完善，希望后面开发者研究一下，给我们提供更好的版本，期待中……】

【相关测试截图如下】：

（1）Thino 笔记中的卡片显示截图
![Thino卡片中的显示【版本2 4 16】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/83cd885b-8793-480e-aed7-72b9771f2971)

（2）Thino笔记在【每日Thino】中得显示效果截图
![每日Thino中卡片显示格式【版本：2 4 16】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/2618973e-0fc1-4e48-81f8-8d208684a690)

（3）Thino 笔记对应日记 md 文件编辑视图下的截图
![Thino测试日记md中编辑视图【版本2 4 16】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/d97394d5-de96-4164-96e4-ab3e8e2790ed)

（4）Thino 笔记对应日记 md 文件阅读视图下的截图
![Thino测试日记md中阅读视图【版本2 4 16】](https://github.com/Quorafind/Obsidian-Thino/assets/99108639/ecbf6880-4862-48b6-ae7f-430233c0adb8)

[a comment by an automated account is not shown]

**Comment by windily-cloud** [comment]
相关issue:

- #29 和 #99  不希望使用`<br>`展示换行关系，因为这对子列表不兼容
- #108 提到使用`<br>`对于块格式的内容无法渲染
- #122 提到 `<br>`会导致内容过长，影响阅读

此外，讨论群里也有相关表达对`<br>`换行的不满：

![图片](https://github.com/Quorafind/Obsidian-Thino/assets/30640638/4e0b89f4-eef4-45d1-beed-6eb8c24316be)

@whong1618 请问这些用户的看法是否合理，如果能处理这些问题会积极采纳之前的方案。
